File: frontend/lib/models/expense.dart
class Expense {
  int id;
  String name;
  String description;
  double amount;
  String supplier;
  int status;
  String date;
  int adminID;


  Expense({
    this.id = -1,
    required this.name,
    required this.description,
    required this.amount,
    required this.supplier,
    required this.status,
    required this.date,
    required this.adminID,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    for (var key in json.keys) {
      if (json[key] == null) {
        json[key] = '';
      }
    }
    return Expense(
      id: json['id'].toInt(),
      name: json['product_name'],
      description: json['description'],
      amount: json['amount'].toDouble(),
      supplier: json['supplier'],
      status: json['status'].toInt(),
      date: json['date'],
      adminID: json['admin_member_id'].toInt(),
    );
  }

}

File: server/src/models/expense.ts
import { DataTypes } from "sequelize";
import db from "../../db/connection";

export const Expense = db.define('expense', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    product_name: {
        type: DataTypes.STRING,
        allowNull: false
    },
    description: {
        type: DataTypes.STRING,
        defaultValue: null
    },
    amount: {
        type: DataTypes.FLOAT,
        allowNull: false
    },
    supplier: {
        type: DataTypes.STRING,
        defaultValue: null
    },
    //1 = pendiente, 2 = aprobado, 3 = anulado
    status: {
        type: DataTypes.INTEGER,
        allowNull: false
    },
    date: {
        type: DataTypes.DATE,
        allowNull: false
    },
    admin_member_id: {
        type: DataTypes.INTEGER,
        allowNull: false
    }
},
    {
        freezeTableName: true,
        
    })